src/Infrastructure/LeanCode.AuditLogs/ChangedEntitiesExtractor.cs

```diff
@@ -0,0 +1,58 @@
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.EntityFrameworkCore;
+
+namespace LeanCode.AuditLogs;
+
+public static class ChangedEntitiesExtractor
+{
+    private static readonly JsonSerializerOptions Options =
+        new()
+        {
+            Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+            ReferenceHandler = ReferenceHandler.IgnoreCycles,
+            WriteIndented = false,
+        };
+
+    public static IReadOnlyList<EntityData> Extract(DbContext dbContext)
+    {
+        return dbContext.ChangeTracker
+            .Entries()
+            .Where(e => e.State != EntityState.Unchanged && e.State != EntityState.Detached)
+            .Select(
+                e =>
+                    new EntityData(
+                        // TODO: FIXME, I fail with owned entities
```

Comment:
Fixed, hopefully?

Reply:
Well, I guess `fail` is not a right word here. The key name is not extracted for owned entities

Reply:
So, we still need to fix something before releasing, right?

Reply:
Considering our goal here, I'd say no. We discussed it some time ago and we want to focus on the aggregates logging here (which works fine, they get logged with owned entities for which we cannot extract key). There is no meaningful log for those entities that are owned by some other entities, which I don't find being a problem, since they do not have a meaning without root entity

Reply:
Okay, so let's get rid of the TODO then? :D

Reply:
Yeah, I'll replace it with more suitable comment